dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -194,14 +201,40 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                context = await f.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
+                var functionDetails = f.Describe();
+                string? renderedPrompt = null;
+
+                SemanticFunction? semanticFunction = f as SemanticFunction;
+
+                if (semanticFunction is not null)
+                {
+                    semanticFunction.AddDefaultValues(context.Variables);
+                    renderedPrompt = await semanticFunction._promptTemplate.RenderAsync(context, cancellationToken).ConfigureAwait(false);
```

Comment:
Will `RenderAsync` get called again when the function is invoked?

If yes, this could be a problem because the prompt could contain functions and the values those functions return could change.